### PR TITLE
[APP-3958] Maplibre component cleanup

### PR DESCRIPTION
**Pull request by micheal-parks**
### Overview

This PR cleans up the `<MapLibre>` component and makes it more flexible for debugging use. I made a quick attempt to unit test but since Maplibre uses WebGL it was becoming more difficult than what it's worth at the moment.

### Changes

* Upgrade MapLibre version.
* Extract map controls to new component for composability.
* Replace events with callback props.
* Remove `!` usage.
* Make `<Marker>` more efficient / easy to use.
* Add example for a map following a marker.

More details in comments.

### CODE REVIEW

File: packages/blocks/src/lib/maplibre/controls.svelte [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<!--
```
Comment (micheal-parks) [inline packages/blocks/src/lib/maplibre/controls.svelte:1]: Extracted out of the `<MapLibre>` component

File: packages/blocks/src/lib/navigation-map/index.svelte [hunk 2]
```diff
@@ -71,7 +71,6 @@ $: $envStore = environment;
 <Map
   bind:map
   {baseGeoPose}
-  on:create
```
Comment (micheal-parks) [inline packages/blocks/src/lib/navigation-map/index.svelte:74]: Not used in app or rdk - so a safe breaking change

File: packages/blocks/src/routes/following-marker.svelte [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<script lang="ts">
```
Comment (micheal-parks) [inline packages/blocks/src/routes/following-marker.svelte:1]: This example is basically how we want to use MapLibre in the movement sensor debug card.

File: packages/blocks/src/lib/maplibre/index.svelte [hunk 4]
```diff
@@ -102,22 +116,20 @@ onMount(() => {
...
   map.on('move', handleMove);
   map.on('resize', handleResize);
   map.on('style.load', handleCreate);
-});
 
-onDestroy(() => {
-  map?.off('move', handleMove);
-  map?.off('resize', handleResize);
-  map?.off('style.load', handleCreate);
+  return () => {
```
Comment (micheal-parks) [inline packages/blocks/src/lib/maplibre/index.svelte:128]: I've been avoiding `onDestroy` when possible. It runs server side whereas the mount cleanup function does not.

**Review by ethanlook — APPROVED** [review]
Awesome! Did you try using this with locally running RDK?

**Comment by micheal-parks** [comment]
Yep! All behavior looks unchanged 👍

**Review by zaporter-work — COMMENTED** [review]
Bit late to the review, but this is great!

**Pull request merged**